clang/lib/Driver/ToolChains/Darwin.cpp

```diff
@@ -2990,26 +2996,18 @@ static bool sdkSupportsBuiltinModules(
     return false;
 
   VersionTuple SDKVersion = SDKInfo->getVersion();
-  switch (TargetPlatform) {
+  switch (SDKInfo->getOS()) {
   // Existing SDKs added support for builtin modules in the fall
   // 2024 major releases.
-  case Darwin::MacOS:
+  case llvm::Triple::MacOSX:
     return SDKVersion >= VersionTuple(15U);
-  case Darwin::IPhoneOS:
-    switch (TargetEnvironment) {
-    case Darwin::MacCatalyst:
```

Comment:
How is it handled in the new version? I don't see a special case for MacCatalyst in the new code, though I could have missed it.

Reply:
The new version only uses the OS of the SDK, it doesn't matter what the target is anymore and there is no Mac Catalyst SDK, only macOS.

Reply:
I should also say that Mac Catalyst is explicitly tested in clang/test/Driver/darwin-builtin-modules.c